### ISSUE TITLE
Patch 1.31

### ISSUE THREAD

**Issue by fe3dback** [wurstscript/WurstScript#842]
Hi, any plans to implement 1.31 support? (new stdlib functions, working with extracted map folders, fix executable path, etc..)?

**Comment by Frotty** [comment]
Sure, see https://github.com/wurstscript/WurstScript/issues/841

> new stdlib functions

only makes sense when it's not PTR, or there is a PTR branch.

> working with extracted map folders

What's there to work with? We already had access to all files in the map before.

> fix executable path

Hasn't changed, runmap target can be configured via run-args.

**Comment by peq** [comment]
> working with extracted map folders

Might be useful if you want to keep your map file in Git. Then you only have to commit the files that really changed inside the map.

**Comment by fe3dback** [comment]
> What's there to work with? We already had access to all files in the map before.
in PTR, world editor can work natively with extracted maps. I don`t tested how game work with extracted maps, but if it can - this is good chance to optimize map compiling/saving process. Wurst compiler can skip all steps to generate all map files, and save only code to few specific *.w3* files, without full MPQ repack.

> Hasn't changed, but yes, runmap target changed and can be configured via settings.json
good, how to configure wc3 executable path?

**Comment by Frotty** [comment]
> this is good chance to optimize map compiling/saving process. Wurst compiler can skip all steps to generate all map files, and save only code to few specific .w3 files, without full MPQ repack.

Only the war3map.j and war3map.w3i are generated. Those steps including packing the mpq usually take <5% of the build duration. I don't think it's worth optimizing that.

@peq mentioned a valid point for version control though.

**Comment by fe3dback** [comment]
I can't run map from wurst launcher in PTR, because Warcraft3 data folder hardcoded here:
https://github.com/wurstscript/WurstScript/blob/master/de.peeeq.wurstscript/src/main/java/de/peeeq/wurstio/languageserver/requests/RunMap.java as

```text
\My Documents\Warcraft III
```

What i try:
- make symlink from `Warcraft III Public Test` to `Warcraft III` (launcher successfully copy map to this directory, but game not see map, because spaces in directory name, see https://github.com/wurstscript/WurstScript/issues/845#issuecomment-491023533)
- setup path in VSCode `wurst.mapDocumentPath` to something like this `C:\Users\fe3db\Documents\wc3ptr` instead of `C:\Users\fe3db\Documents\Warcraft III Public Test` (in user options and in .wscode/settings.json), but wurstscript launcher ignore this completly.
- add compileArgs (runmapTarget) to "wurst_run.args", like this:
```
Following are the default enabled arguments:
-runmapTarget "C:\Users\fe3db\Documents\wc3ptr"
-runcompiletimefunctions
-injectobjects
-stacktraces
```

but in this case command "run map" crash instantly without any errors in logs. IDE display "Internal error" message on bottom-right corner. (wurst debug mode is ON, wurst hide exceptions is OFF)

In code this logic defined here:
https://github.com/wurstscript/WurstScript/blob/20015e388b9e2afd6a21064d20478d45da9a6b87/de.peeeq.wurstscript/src/main/java/de/peeeq/wurstio/languageserver/requests/RunMap.java#L225

So any ideas, how to run map in PTR?

**Comment by peq** [comment]
You can try the "lua" branch, which has experimental Lua support and allows configuring this folder.

**Comment by Frotty** [comment]
Closed by 2a7d50bcfe91290e191af1ec992dbe0eda3d6d0d and 65e851c7513eeb4cff51a4670e56663486a2736f
